trunk/cl_parse.c

```diff
@@ -1489,7 +1492,10 @@ void CL_ParseServerMessage (void)
 				cl.completed_time = cl.mtime[0];	//joe: intermission bugfix
 				PrintFinishTime();
 			}
-			cl.intermission = 2;
+			if (cls.demoplayback && cl_demorewind.value)
+				cl.intermission = 1;
```

Comment:
This line should assign zero I think.  Rewinding back from the finale leaves the intermission screen visible.

Reply:
Oh, I see.  For episode ending maps this is right, but for `end` it is not.  Episode ends have both an `intermission` and `finale` message, but `end` just has `finale`.  Tricky.

Reply:
That's the idea. At least on maps like e1m7, the intermission (`cl.intermision == 1`) is followed by the finale text (`cl.intermission == 2`). So playing it backwards should go from `2` to `1` then back to `0`. Unless you are thinking of a different use case?

Reply:
Ah I see, didn't check `end`. Would need to store previous value (or even a full stack) to fix every edge case I guess. I'm more than happy with this for now, 99% of maps should work already.